### PR TITLE
testing(sample): add presubmit builds against head

### PR DESCRIPTION
**Pull request by tmatsuo**
This should be non-destructive change. I'll add a commit for actually
adding a new type of periodic build.

### CODE REVIEW

**Comment by tmatsuo** [comment]
@dizcology  FYI

**Comment by tmatsuo** [comment]
I'll also run a periodic build manually against my branch.

**Comment by tmatsuo** [comment]
lol it doesn't work ([log](https://source.cloud.google.com/results/invocations/7129972e-81e3-40ee-a6ab-2c574e05efea/log)):

```
--- IMPORTANT IMPORTANT IMPORTANT ---
Now we rewind the repo back to the latest release...
Previous HEAD position was 26d54e3 fixes the script path
HEAD is now at a11ac13 chore: release v2.4.0 (#296)
The current head is: 
a11ac1373cd1996089620d8b72b4b4d1b3f1812f
--- IMPORTANT IMPORTANT IMPORTANT ---
github/python-pubsub/.kokoro/test-samples.sh: line 40: /tmpfs/src/github/python-pubsub/.kokoro/test-samples-impl.sh: No such file or directory
cleanup
```

**Comment by tmatsuo** [comment]
Manual periodic build suceeded ([log](https://source.cloud.google.com/results/invocations/fa37b01c-fcae-4b97-8ddf-d081233adc1b/log)), and I see the following in the log:

```
--- IMPORTANT IMPORTANT IMPORTANT ---
Now we rewind the repo back to the latest release...
Previous HEAD position was 5d39581 fix a bug, also add a new wrapper
HEAD is now at a11ac13 chore: release v2.4.0 (#296)
The current head is:
a11ac1373cd1996089620d8b72b4b4d1b3f1812f
--- IMPORTANT IMPORTANT IMPORTANT ---
```

File: .kokoro/test-samples.sh [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
-# Copyright 2020 Google LLC
+# Copyright 2021 Google LLC
```
Comment (anguillanneuf) [inline .kokoro/test-samples.sh:2]:
```suggestion
# Copyright 2020 Google LLC
```

File: .kokoro/test-samples-impl.sh [hunk 2]
```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+# Copyright 2020 Google LLC
```
Comment (anguillanneuf) [inline .kokoro/test-samples-impl.sh:2]:
```suggestion
# Copyright 2021 Google LLC
```

**Review by anguillanneuf — APPROVED** [review]
LGTM. Just some nits.

**Comment by tmatsuo** [comment]
The manual run of py-3.8 presubmit-head build passed:
https://source.cloud.google.com/results/invocations/de713930-2200-4fd7-b7a7-a71225313129/targets

**Pull request merged**